File: keevo/backend/src/main/java/com/keevo/identity/auth/domain/model/AuthTokens.java
package com.keevo.identity.auth.domain.model;

import java.util.UUID;

/**
 * AuthTokens — Domain model returned after successful authentication.
 *
 * <p>Pure Java record — NO Spring, JPA, or framework imports.
 *
 * @param accessToken              RS256-signed JWT (24h expiry)
 * @param refreshToken             opaque random bytes, base64url-encoded (30-day expiry)
 * @param expiresIn                access token lifetime in seconds
 * @param userId                   authenticated user's UUID (used by REST adapter to build response)
 * @param tenantId                 tenant schema name (kv_xxxxxx) for client routing
 * @param role                     user role string (OWNER | EMPLOYEE)
 * @param storeId                  assigned store UUID string (null for OWNER) — Story 3.5
 * @param passwordChangeRequired   true if employee must change password — Story 3.5
 */
public record AuthTokens(
    String accessToken,
    String refreshToken,
    long   expiresIn,
    UUID   userId,
    String tenantId,
    String role,
    String storeId,
    boolean passwordChangeRequired
) {
    /** Backward-compatible constructor for OWNER tokens (storeId=null, passwordChangeRequired=false). */
    public AuthTokens(String accessToken, String refreshToken, long expiresIn,
                      UUID userId, String tenantId, String role) {
        this(accessToken, refreshToken, expiresIn, userId, tenantId, role, null, false);
    }
}

File: keevo/app/lib/features/auth/domain/model/auth_tokens.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'auth_tokens.freezed.dart';

/// AuthTokens — Domain model for a successful authentication response.
///
/// Immutable Freezed value object — auto-generates equality, hashCode,
/// copyWith and toString.
///
/// AC1: returned after successful login or token refresh.
/// AC3: refreshToken is used to silently renew the access token.
@freezed
class AuthTokens with _$AuthTokens {
  const factory AuthTokens({
    /// RS256-signed JWT (24h expiry). Sent as Authorization: Bearer header.
    required String accessToken,

    /// Opaque refresh token (30-day expiry). Never sent in regular requests.
    required String refreshToken,

    /// Authenticated user's UUID.
    required String userId,

    /// Tenant schema name (kv_xxxxxx) for multi-tenant routing.
    required String tenantId,

    /// User role: 'OWNER' or 'EMPLOYEE'.
    required String role,

    /// Access token lifetime in seconds (defaults to 86400 = 24h).
    required int expiresIn,

    /// Assigned store UUID (null for OWNER) — Story 3.5.
    String? storeId,

    /// True if employee must change password on first login — Story 3.5.
    @Default(false) bool passwordChangeRequired,
  }) = _AuthTokens;
}
